Fix/ion tests await and update transmute did-key dep

nklomp opened the pull request:
## What issue is this PR fixing
Intermittent timeout exceptions occur on ION test when running locally as reported by @nickreynolds  I had a hard time reproducing it using several environments, operating systems and node versions and actually managed to only get the error one time.

Looking at the test code it potently could happen because certain corner cases didn't await the matching methods which are async. Ran the tests many times over, and could not reproduce it anymore

Also removed the dep to transmute's did-key-jwk/secp256k1 support. The ION provider was using an older version of the dep than used elsewhere. The only reason was because it used 2 methods to convert secp256k private and public keys to JWKs. Since the structure of the package has been completely overhauled in recent versions, and these implementations changed, decided to copy the respective methods and use some previous transitive deps directly. 

Potential fixes #1066 . But I think that problem is not related to the mismatch in package versions

## What is being changed
See above

## Quality
Check all that apply:
* [X] I want these changes to be integrated
* [x] I successfully ran `yarn`, `yarn build`, `yarn test`, `yarn test:browser` locally.
* [x] I allow my PR to be updated by the reviewers (to speed up the review process).
* [x] I added unit tests.
* [x] I added integration tests.
* [ ] I did not add automated tests because _________, and I am aware that a PR without tests will likely get rejected.

mirceanis commented on packages/did-provider-ion/src/functions.ts:
```diff
@@ -48,9 +55,75 @@ export const toIonPrivateKeyJwk = (privateKeyHex: string): JwkEs256k => {
+  let key = publicKeyHex;
+  const compressedHexEncodedPublicKeyLength = 66;
+  if (publicKeyHex.length === compressedHexEncodedPublicKeyLength) {
+    const keyBin = secp256k1.publicKeyConvert(
+      Buffer.from(publicKeyHex, 'hex'),
+      false
+    );
+    key = Buffer.from(keyBin).toString('hex');
```
We already have @ethersproject/* in the dependency tree (usually) so It would be better to use that for public key conversions instead of a new dependency.

Using this import `import { computePublicKey } from '@ethersproject/signing-key'`,
I think this would work to convert the key to uncompressed hex encoding:
```suggestion
    key = computePublicKey(`0x${publicKeyHex}`, false).substring(2)
```

This also removes the need for `Buffer` which requires special treatment on web.

nklomp replied:
Removed Buffer

mirceanis commented on packages/did-provider-ion/src/functions.ts:
```diff
@@ -48,9 +55,75 @@ export const toIonPrivateKeyJwk = (privateKeyHex: string): JwkEs256k => {
+    .update(canonicalize(copy)!, 'utf-8')
+    .digest();
+
+  return base64url.encode(Buffer.from(digest));
```
```suggestion
  return u8a.toString(digest, 'base64url')
```

We prefer to avoid Buffer since it requires special treatment on web.
Perhaps with this change we also don't need `base64url` dependency anymore?

mirceanis commented on packages/did-provider-ion/src/functions.ts:
```diff
@@ -48,9 +55,75 @@ export const toIonPrivateKeyJwk = (privateKeyHex: string): JwkEs256k => {
+  delete copy.d;
+  delete copy.kid;
+  delete copy.alg;
+  const digest = crypto
```
In other parts of the codebase we use `import { hash } from '@stablelib/sha256'` to compute sha256.

What do you think about replacing the use of `crypto`, since it requires special treatment on web?

nklomp replied:
Updated it everywhere

mirceanis reviewed (changes requested):
Thanks for adding this!

I have some suggestions that would replace some of the dependencies with others that are already used in other parts of the codebase.
Please take a look and let me know what you think.

Also, my delayed review allowed a conflict to appear that has to be solved before we can merge. Sorry about that.

nklomp commented:
Removed some packages and also removed usage of crypto and Buffer.
Updated to the latest ion-sdk. Should be browser compatible, but didn't get around to testing it yet (and probably won't this week)

mirceanis reviewed (approved):
Looks great, thank you for the prompt edits!
